As in the simple linear regression the GPA was calculated using SAT only. 

GPA cannot be predicted solely by students SAT score but also by income, gender and so on.

Good models require multiple regressions,in order to address the higher complexity of problems.

The more variables we have the more factors you are considering in a model.

Multiple regression equation:

yhat = b0 + b1x1 + b2x2 + ..... + bk.xk

yhat = inferred value

b0 = intercept

b1,b2,b3 ... bk = the independent variables

After 3 dimensions, there is no visual way to represent the data

As we know from OLS we want the least SSE so to descrese it we need to adjust the SSR. As they are inversely proportional to each other.

Multiple regression is all about best fitting the model.

# **Adjusted R squared**

1. The R-squared measures how much of the total variability is explained by our model

2. Multiple regression are always better than simple ones, as with each additional variable you add, the explanatory power may only increase or stay the same

**Adjusted R squared < R squared**


**Multiple linear regression and adjusted R-squared**

**Import the relevant libraries**

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
import seaborn as sns
sns.set()

**Load the data**

In [2]:
data = pd.read_csv("dataset/Multiple_linear_regression.csv")
data.head()

SAT   GPA  Rand 1,2,3
0  1714  2.40           1
1  1664  2.52           3
2  1760  2.54           3
3  1685  2.74           3
4  1693  2.83           2

In this dataset we have an additional variable Rand 1,2,3 which is generated randomly.

In [3]:
data.describe()

SAT        GPA  Rand 1,2,3
count    84.000000  84.000000   84.000000
mean   1845.273810   3.330238    2.059524
std     104.530661   0.271617    0.855192
min    1634.000000   2.400000    1.000000
25%    1772.000000   3.190000    1.000000
50%    1846.000000   3.380000    2.000000
75%    1934.000000   3.502500    3.000000
max    2050.000000   3.810000    3.000000

**Create multiple regression**

In [4]:
y = data['GPA']
x1 = data[['SAT','Rand 1,2,3']]

In [5]:
x = sm.add_constant(x1)
results = sm.OLS(y,x).fit()

C:\Users\priya\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:2542: FutureWarning: Method .ptp is deprecated and will be removed in a future version. Use numpy.ptp instead.
  return ptp(axis=axis, out=out, **kwargs)


In [6]:
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    GPA   R-squared:                       0.407
Model:                            OLS   Adj. R-squared:                  0.392
Method:                 Least Squares   F-statistic:                     27.76
Date:                Fri, 29 May 2020   Prob (F-statistic):           6.58e-10
Time:                        18:03:55   Log-Likelihood:                 12.720
No. Observations:                  84   AIC:                            -19.44
Df Residuals:                      81   BIC:                            -12.15
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.2960      0.417      0.710      0.480      -0.533       1.125
SAT            0.0017      0.000      7.432      0.000       0.001       0.002
Rand 1,2,3    -0.0083      0.027     -0.304      0.762      -0.062       0.046
==============================================================================
Omnibus:                       12.992   Durbin-Watson:                   0.948
Prob(Omnibus):                  0.002   Jarque-Bera (JB):               16.364
Skew:                          -0.731   Prob(JB):                     0.000280
Kurtosis:                       4.594   Cond. No.                     3.33e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 3.33e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

As the Adj. R-squared is less than R-squared we were penalized for adding an additional variable that had no strong explanatory power.

The model shows the variables which are not useful. 

As seen above, the pvalue for Rand1,2,3 is 0.762. We cannot reject the null hypothesis at 76% level. We want a p-value of 0.05.

Hence, the variable 'Rand1,2,3' not only worsens the explanatory power, but is also insignificant.

As the equation changed from 

yhat = 0.275 + 0.0017x1

yhat = 0.296 + 0.0017x1 - 0.0083x2

The bias of this variable is reflected into the coefficients of the others.


**Adjusted R squares is the basis for comparing models**

1. Same dependent variable(y)
2. Same dataset



# Assessing overall significance of the model

**F - Statistics**

It is used for testing the overall significance of the model

F-test:

H0: b1=b2=b3.... =bk=0
H1: at least one bi not equal to 0

If all betas are 0, then none of the X's matter => our model has not merit

The F-statistics is 27.76 and 

Prob(F-statistics) = 6.58e-11 which is ~0.000

Hence we say the overall model is significant.

**The lower the F-statistic the closer to a non-significant model.**

